File: custom-controls/src/main/scala/scalafx/scene/control/custom/YieldingSlider.scala
package scalafx.scene.control.custom

import javafx.scene.{control => jfxsc}

import scalafx.scene.control.Slider

object YieldingSlider {
  implicit def sfxSlider2jfx(v: YieldingSlider) = v.delegate
}

class YieldingSlider(override val delegate: jfxsc.custom.YieldingSlider = new jfxsc.custom.YieldingSlider) extends Slider {

   /** Constructs a Slider control with the specified slider min, max and current value values. */
   def this(min:Double, max:Double, value:Double) {
     this(new jfxsc.custom.YieldingSlider(min, max, value))
   }
 }

File: custom-controls/src/main/java/javafx/scene/control/custom/YieldingSlider.java
package javafx.scene.control.custom;

import javafx.event.EventHandler;
import javafx.fxml.FXMLLoader;
import javafx.scene.control.Slider;
import javafx.scene.input.MouseEvent;

import java.io.IOException;

public class YieldingSlider extends Slider {

    public YieldingSlider() {
        addEventFilter(MouseEvent.MOUSE_PRESSED, new EventHandler<MouseEvent>() {
            @Override
            public void handle(final MouseEvent event) {
                lastTimeMousePressed = System.currentTimeMillis();
            }
        });
    }

    public YieldingSlider(final double min, final double max, final double value) {
        this();
        setMin(min);
        setMax(max);
        setValue(value);
    }

    private long lastTimeMousePressed = 0;

    public boolean mouseWasPressedWithinLast(final long t) {
        return (System.currentTimeMillis() - lastTimeMousePressed) <= t;
    }
}
